source/rpt/rpt_cell_reconstruction.cc

```diff
@@ -0,0 +1,454 @@
+#include <deal.II/grid/cell_id.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/manifold_lib.h>
+
+#include <deal.II/numerics/data_component_interpretation.h>
+#include <deal.II/numerics/data_out.h>
+
+#include <rpt/particle_detector_interactions.h>
+#include <rpt/rpt.h>
+#include <rpt/rpt_cell_reconstruction.h>
+
+#include <fstream>
+
+template <int dim>
+RPTCellReconstruction<dim>::RPTCellReconstruction(
+  RPTCalculatingParameters &rpt_parameters)
+  : parameters(rpt_parameters)
+{
+  // Seed the random number generator
+  srand(parameters.rpt_param.seed);
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::execute_cell_reconstruction()
+{
+  assign_detector_positions(); // Assign detector position & create detectors
+  read_counts(); // Read reconstruction counts of unknown particle positions
```

Comment:
Read **experimental counts** of unknown particle position I think can be a better explanation!